domain/account/usecase/create_account_test.go

```diff
@@ -4,72 +4,102 @@ import (
 	"errors"
 	"testing"
 
+	"github.com/francisleide/ChallengeGo/domain/account"
 	"github.com/francisleide/ChallengeGo/domain/account/usecase"
 	"github.com/francisleide/ChallengeGo/domain/entities"
 	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"golang.org/x/crypto/bcrypt"
 )
 
-type repoMock struct {
-	Account entities.Account
-}
-
-func (r *repoMock) ListAllAccounts() ([]entities.Account, error) {
-	return nil, nil
-}
-
-func (r *repoMock) FindOne(CPF string) (entities.Account, error) {
-	return entities.Account{
-		Balance: 100.0,
-		CPF:     "12345679210",
-	}, nil
-}
-
-func (r *repoMock) FindByID(accountID string) (entities.Account, error) {
-	return entities.Account{}, nil
-}
-func (r *repoMock) UpdateBalance(ID string, balance float64) error {
-	return nil
-}
-func (r *repoMock) InsertAccount(account entities.Account) error {
-	if account.CPF == "12345678911" {
-		return errors.New("the account already exists")
-	}
-	return nil
-}
-
 func TestCreateAccount(t *testing.T) {
-	log := *logrus.NewEntry(logrus.New())
-	var account entities.Account
-	account.CPF = "12345678910"
 	t.Run("CPF does not exist in the bank and the account is created", func(t *testing.T) {
-		r := repoMock{}
-		r.Account = account
-		a := usecase.NewAccountUc(&r, log)
-		var accountInput entities.AccountInput
-		accountInput.CPF = "12345678910"
-		accountInput.Name = "Test"
-		accountInput.Secret = "111"
-		accountReceived, err := a.CreateAccount(accountInput)
-		if err != nil {
-			t.Errorf("Error: %s", err)
-		}
-		if accountReceived.CPF != "12345678910" {
-			t.Errorf("Expected %s and was received  %s", accountInput.CPF, accountReceived.CPF)
+		log := logrus.NewEntry(logrus.New())
+		mockRepo := new(account.MockRepository)
+		accountUC := usecase.NewAccountUc(mockRepo, log)
+		accountInput := entities.AccountInput{
+			CPF:    "47708141001",
+			Secret: "abc123",
+			Name:   "Paulo Saulo",
 		}
+		mockRepo.On("FindOne").Return(entities.Account{}, errors.New(""))
+		mockRepo.On("InsertAccount").Return(nil)
+		accountReceived, err := accountUC.CreateAccount(accountInput)
+		compare := bcrypt.CompareHashAndPassword([]byte(accountReceived.Secret), []byte(accountInput.Secret))
+		assert.Nil(t, err)
+		assert.Equal(t, accountInput.CPF, accountReceived.CPF)
+		assert.Equal(t, accountInput.Name, accountReceived.Name)
+		assert.Nil(t, compare)
```

Comment:
WDYT to take this approach changing `compare` to `err` for better semantic? Like:
```suggestion
		assert.NoError(t, err)
		assert.Equal(t, accountInput.CPF, accountReceived.CPF)
		assert.Equal(t, accountInput.Name, accountReceived.Name)
		err := bcrypt.CompareHashAndPassword([]byte(accountReceived.Secret), []byte(accountInput.Secret)) // or errSecret
		assert.NoError(t, err)
```